src/trie.c

```diff
@@ -84,3 +81,83 @@ int insert_string(char *word, trie_t *t)
         return insert_string(word, t->children[(unsigned)curr]);
     }
 }
+
+
+trie_t *trie_search_end(char* word, trie_t *t)
```

Comment:
`trie_search_end` is a pretty uninformative name here. Let's do something like `trie_get_subtrie`

Reply:
I'll can fix this in my other issue of cleaning up the source code (same thing with all other style issues you mention here, there's already a bunch of formatting fixes in there) @mpcsh